map/mode.h

```diff
@@ -9,96 +9,85 @@
- * @param channel_id - A known channel ID [in]
- * @param bundle - The bundle that the packet will be written into [out]
- * @param msg_id - The msg_id [out]
- * @param new_endpoint - The new endpoint [out]
+ * @param channel_depth - Depth of channel merkle tree[in]
+ * @param endpoint_depth - Depth of endpoint merkle tree[in]
+ * @param psks - Pre-Shared Key set[in]
+ * @param ntru_pks - NTRU public key set[in]
+ * @param payload - The message that is going to send with MAM[in]
+ * @param chid - Channel ID[out]
+ * @param epid - Endpoint ID[out]
+ * @param bundle - The bundle contains the Header and Packets of the current Message[out]
+ * @param msg_id - Unique Message identifier under each channel[out]
+ * @param ch_remain_sk - The number of MSS channel secret keys that haven't been used[out]
+ * @param ep_remain_sk - The number of MSS endpoint secret keys of that haven't been used[out]
  *
  * @return return code
  */
-retcode_t map_announce_endpoint(mam_api_t* const api, tryte_t const* const channel_id,
-                                bundle_transactions_t* const bundle, trit_t* const msg_id,
-                                tryte_t* const new_endpoint_id);
+status_t map_written_msg_to_bundle(const iota_client_service_t* const service, mam_api_t* const api,
+                                   const size_t channel_depth, const size_t endpoint_depth, tryte_t* const chid,
+                                   tryte_t* const epid, mam_psk_t_set_t psks, mam_ntru_pk_t_set_t ntru_pks,
+                                   char const* const payload, bundle_transactions_t** bundle, tryte_t* msg_id,
+                                   uint32_t* ch_remain_sk, uint32_t* ep_remain_sk);
 
 /**
- * Writes a header only bundle on a channel
+ * @brief Write an announcement to bundle.
  *
- * @param api - The API [in,out]
- * @param channel_id - A known channel ID [in]
- * @param bundle - The bundle that the packet will be written into [out]
- * @param msg_id - The msg_id [out]
- *
- * @return return code
- */
-retcode_t map_write_header_on_channel(mam_api_t* const api, tryte_t const* const channel_id,
-                                      bundle_transactions_t* const bundle, trit_t* const msg_id);
-
-/**
- * Writes a header only bundle on an endpoint
- *
- * @param api - The API [in,out]
- * @param channel_id - A known channel ID [in]
- * @param endpoint_id - A known endpoint ID [in]
- * @param bundle - The bundle that the packet will be written into [out]
- * @param msg_id - The msg_id [out]
- *
- * @return return code
- */
-retcode_t map_write_header_on_endpoint(mam_api_t* const api, tryte_t const* const channel_id,
-                                       tryte_t const* const endpoint_id, bundle_transactions_t* const bundle,
-                                       trit_t* const msg_id);
-
-/**
- * Writes a packet on a bundle
+ * Write the announcement of the next possible Channel ID (chid1) or Endpoint ID (epid1). This function would
```

Comment:
What do you mean the next channel/endpoint ID is `possible`?

Reply:
Since the chid1 has several possible values. If the description here is obscure, then I can remove the `possible` here

Reply:
But just return one chid, right? If so, maybe remove `possible` would be more clearer.

Reply:
Sure